common/build_all_function.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+build_all() {
+    types=$1; shift
+    goodcount=0
+    projectcount=0
+    for project in $@; do
+        if [ ! -d $project ]; then continue; fi
+        
+        echo -e "\e[34m========== BUILDING $project ==========\e[39m"
+        cd "$project"
```

Comment:
With the GeNN projects you also need to generate the simulation code:

``genn-buildmodel -i $BOB_ROBOTICS_PATH model.cc`` for ardin_mb
``genn-buildmodel -c -i $BOB_ROBOTICS_PATH model.cc`` for stone_cx (cos it doesn't use GPU)

Reply:
There's also the question of what version of GeNN it builds against...

Reply:
Yeah this is why I haven't made a build_all.sh for the "projects" yet -- just for the examples and the tools.

I guess our options for the version of GeNN we use are:
* Whatever version is lurking on our machines already
* The latest version from master
* The latest release
In any case I agree it would be helpful to report this version (inc. the commit hash).

Reply:
Ahh ok - I wasn't entirely clear on how integrated this is. Checking out GeNN master and building against that would definitely be the best thing to do. In GeNN's Jenkinsfile we use curl to download the tar gz of libraries we want to use from github, unzip them and point environment variables at them. I can have a go at that in another PR if you'd like?

Reply:
Sure, that'd be awesome :-)